nx/lib/nx/defn/grad.ex

```diff
@@ -1022,4 +1045,22 @@ defmodule Nx.Defn.Grad do
   defp up_to(_, _), do: []
 
   defp argsort(list), do: list |> Enum.with_index() |> Enum.sort() |> Enum.map(&elem(&1, 1))
+
+  # copies the lower triangle over the upper triangle of a 2-D square tensor
+  defp copyltu(tensor) do
+    # Trick to yield a matrix where every element below
+    # the main diagonal is 1, and the rest is 0
+    selector = Nx.greater(Nx.iota(tensor, axis: 0), Nx.iota(tensor, axis: 1))
+
+    zeros = Nx.broadcast(0, tensor)
+
+    # Fetch the lower triangle of the matrix without the diagonal
+    lower = Nx.select(selector, tensor, zeros)
+
+    # Get the diagonal
+    diag = Nx.select(identity, tensor, zeros)
```

Comment:
`identity` here is undefined. :)

Reply:
ooops, it went missing after a minor refactor :sweat_smile: